Library/Homebrew/tap.rb

```diff
@@ -753,34 +753,32 @@ def cask_files_by_name
     end
   end
 
-  # Check whether the file has a Ruby extension.
-  sig { params(file: Pathname).returns(T::Boolean) }
-  def ruby_file?(file)
-    file.extname == ".rb"
+  sig { returns(Regexp) }
+  def formula_file_regex
+    @formula_file_regex ||= case formula_dir.basename.to_s
+    when "Formula"
+      %r{^Formula(/[^/]+)+\.rb$}
+    when "HomebrewFormula"
+      %r{^HomebrewFormula(/[^/]+)+\.rb$}
+    else
+      %r{^[^/]+\.rb$}
```

Comment:
I should just switch on all of the paths from the `potential_formula_dirs` method instead of using the basename here.

Reply:
Addressed in https://github.com/Homebrew/brew/pull/18167/commits/62a4734c9026500b77d30d092252b7e1ad59e6cc